File: CSW/JavaScript/221024_최소원_신입사원.js
var fs = require('fs');
var [T, ...inputs] = fs.readFileSync(`/dev/stdin`).toString().trim().split('\n');

let n=0; 

for(let i=0; i<T; i++){
    let N = parseInt(inputs[n]);
    let lists = [];
    for(let j=1; j<=N; j++){
        let temp = inputs[n+j].split(' ').map(e=>parseInt(e));
        lists.push(temp);
    }

    lists.sort((a,b)=> a[0]-b[0]);

    let cnt = 1;
    let temp = lists[0][1];

    for(let j=1; j<lists.length; j++){
        if(temp>lists[j][1]){
            temp =lists[j][1];
            cnt+=1;
        }
    }

    console.log(cnt);
    n += N+1;
}

File: CSW/C,C++/221024_최소원_신입사원.c
#include <stdio.h>

int lists[100000][2];
int T, N, cnt, x;

int main(){

    scanf("%d", &T);

    for(int i=0; i<T; i++){
        scanf("%d", &N);
        cnt = 1;
        for(int j = 0; j<N; j++){
            int a, b;
            scanf("%d %d", &a, &b);
            lists[a-1][0] = a;
            lists[a-1][1] = b;
        }
        x = lists[0][1];
        for(int j=1; j<N; j++){
            if(x>lists[j][1]){
                x=lists[j][1];
                cnt+=1;
            }
        }
        printf("%d\n", cnt);
    }
    return 0;
}